spec/index.bs

```diff
@@ -2473,30 +2473,35 @@ dictionary GPUBindGroupLayoutDescriptor : GPUObjectDescriptorBase {
 A {{GPUBindGroupLayoutEntry}} describes a single shader resource binding to be included in a {{GPUBindGroupLayout}}.
 
 <script type=idl>
+typedef [EnforceRange] unsigned long GPUShaderStageFlags;
+interface GPUShaderStage {
+    const GPUFlagsConstant VERTEX   = 0x1;
+    const GPUFlagsConstant FRAGMENT = 0x2;
+    const GPUFlagsConstant COMPUTE  = 0x4;
+};
+
 dictionary GPUBindGroupLayoutEntry {
     required GPUIndex32 binding;
     required GPUShaderStageFlags visibility;
-    required GPUBindingType type;
-
-    // Used for uniform buffer and storage buffer bindings. Must be undefined for other binding types.
-    boolean hasDynamicOffset;
 
-    // Used for uniform buffer and storage buffer bindings. Must be undefined for other binding types.
-    GPUSize64 minBufferBindingSize;
+    // Used for uniform buffer and storage buffer bindings. Must be null for other binding types.
```

Comment:
We should take this opportunity to move this stuff from comments to real prose. The reason it was comments was because this dictionary was so difficult to understand in the first place (understanding which fields were used for which binding types). Now that we're solving that problem, we can simply say just below here, "Exactly one of .buffer, .sampler, .texture, or .storageTexture must be specified. Which is specified, and value of that sub-entry, determines what type of resource can be bound. (Note: a value of `{}` is valid if the type has no required members.)`